From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-reduce-noise-in-show_mem-for-lowmem-allocations-update-fix
Date: Tue Aug 30 06:53:22 PM PDT 2022

fix build, remove unneeded __show_mem() arg

lib/show_mem.c: In function '__show_mem':
lib/show_mem.c:17:9: error: too many arguments to function 'show_free_areas'
   17 |         show_free_areas(filter, nodemask, max_zone_idx);
      |         ^~~~~~~~~~~~~~~
In file included from lib/show_mem.c:8:
./include/linux/mm.h:1842:28: note: declared here
 1842 | static void __maybe_unused show_free_areas(unsigned int flags, nodemask_t *nodemask)
      |                            ^~~~~~~~~~~~~~~

Cc: Christoph Hellwig <hch@infradead.org>
Cc: Dan Carpenter <dan.carpenter@oracle.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Mel Gorman <mgorman@suse.de>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/mm.h |    4 ++--
 lib/show_mem.c     |    4 ++--
 mm/oom_kill.c      |    2 +-
 mm/page_alloc.c    |    2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

--- a/include/linux/mm.h~mm-reduce-noise-in-show_mem-for-lowmem-allocations-update-fix
+++ a/include/linux/mm.h
@@ -2583,10 +2583,10 @@ extern int __meminit init_per_zone_wmark
 extern void mem_init(void);
 extern void __init mmap_init(void);
 
-extern void __show_mem(unsigned int flags, nodemask_t *nodemask, int max_zone_idx);
+extern void __show_mem(unsigned int flags, nodemask_t *nodemask);
 static inline void show_mem(unsigned int flags, nodemask_t *nodemask)
 {
-	__show_mem(flags, nodemask, MAX_NR_ZONES - 1);
+	__show_mem(flags, nodemask);
 }
 extern long si_mem_available(void);
 extern void si_meminfo(struct sysinfo * val);
--- a/lib/show_mem.c~mm-reduce-noise-in-show_mem-for-lowmem-allocations-update-fix
+++ a/lib/show_mem.c
@@ -8,13 +8,13 @@
 #include <linux/mm.h>
 #include <linux/cma.h>
 
-void __show_mem(unsigned int filter, nodemask_t *nodemask, int max_zone_idx)
+void __show_mem(unsigned int filter, nodemask_t *nodemask)
 {
 	pg_data_t *pgdat;
 	unsigned long total = 0, reserved = 0, highmem = 0;
 
 	printk("Mem-Info:\n");
-	show_free_areas(filter, nodemask, max_zone_idx);
+	show_free_areas(filter, nodemask);
 
 	for_each_online_pgdat(pgdat) {
 		int zoneid;
--- a/mm/oom_kill.c~mm-reduce-noise-in-show_mem-for-lowmem-allocations-update-fix
+++ a/mm/oom_kill.c
@@ -461,7 +461,7 @@ static void dump_header(struct oom_contr
 	if (is_memcg_oom(oc))
 		mem_cgroup_print_oom_meminfo(oc->memcg);
 	else {
-		__show_mem(SHOW_MEM_FILTER_NODES, oc->nodemask, gfp_zone(oc->gfp_mask));
+		__show_mem(SHOW_MEM_FILTER_NODES, oc->nodemask);
 		if (should_dump_unreclaim_slab())
 			dump_unreclaimable_slab();
 	}
--- a/mm/page_alloc.c~mm-reduce-noise-in-show_mem-for-lowmem-allocations-update-fix
+++ a/mm/page_alloc.c
@@ -4329,7 +4329,7 @@ static void warn_alloc_show_mem(gfp_t gf
 	if (!in_task() || !(gfp_mask & __GFP_DIRECT_RECLAIM))
 		filter &= ~SHOW_MEM_FILTER_NODES;
 
-	__show_mem(filter, nodemask, gfp_zone(gfp_mask));
+	__show_mem(filter, nodemask);
 }
 
 void warn_alloc(gfp_t gfp_mask, nodemask_t *nodemask, const char *fmt, ...)
_
